src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,233 @@
+        prob_mutate # Mutation probability, usually as small as 0.01 (1%)
+)
+```
+
+Then, you need to create an `EEnvironment` object, the important parameter is `fitness_func`, which is the function name that we declared above:
+
+```py
+from qoop.evolution.environment import EEnvironment
+env = EEnvironment(
+     metadata = env_metadata,
+     fitness_func = fitnessW,
+)
+```
+
+The object `EEnvironment` has other parameters such as `selection_func`, `crossover_func`, `mutate_func`, `threshold_func` which can be imported from below sub-module:
+
+```py
+from qoop.evolution import crossover, mutate, selection, threshold
+```
+
+or defined by yourself.
+
+|   Parrameter   |             Default             |                                                        Function type                                                         |
+| :------------: | :-----------------------------: | :--------------------------------------------------------------------------------------------------------------------------: |
+| selection_func |   selection.elitist_selection   |                 $f$: [qiskit.QuantumCircuit] $\times$ $\mathbb{R}^{N}$ $\rightarrow$ [qiskit.QuantumCircuit]                 |
+| crossover_func |  crossover.onepoint_crossover   | $f$: qiskit.QuantumCircuit $\times$ qiskit.QuantumCircuit $\rightarrow$ qiskit.QuantumCircuit $\times$ qiskit.QuantumCircuit |
+|  mutate_func   |     mutate.layerflip_mutate     |                                $f$: qiskit.QuantumCircuit $\rightarrow$ qiskit.QuantumCircuit                                |
+| threshold_func | threshold.compilation_threshold |                                             $f$: $\mathbb{R} \rightarrow$ {0,1}                                              |
+
+_**Table 2**: Parameter functions for GA-QAS._
+
+# Step 3. Run GA-QAS
+
+You can call the method `evol()` to start running GA-QAS.
+
+```py
+env.evol(
+   verbose = 1,
+   auto_save = True
+)
+```
+
+There are only two parameters: the first is verbose (1 means print % process, 0 means no print), and the second is the saving option.
+The saving result's folder is detailed on [GA-QAS: folder result](https://github.com/vutuanhai237/qoop/wiki/GA%E2%80%90QAS:-folder-result).
+
+# Step 4. Plot result
+
+Currently, we support plotting fitness values against a number of generations. In the future, we will develop more presentation plots.
+
+```py
+env.plot()
+```
+
+<figure>
+    <img src='/public/images/ga-qas/ga-qas3.png' style="margin: auto" alt='description' />
+    <figcaption>Fig 3. Fitness values versus number of generations</figcaption>
+</figure>
+
+The result is saved in a folder; the default folder name is based on the fitness function name. We care about a file named `best_circuit.qpy`, which is our final solution. Then, we can load it by $\langle qo|op\rangle$ and put it into fitness again to test:
+
+```py
+from qoop.backend import utilities
+best_circuit = utilities.load_circuit("Path to best_circuit.qpy")
```

Comment:
```suggestion
best_circuit = utilities.load_circuit("./best_circuit.qpy")
```

Reply:
The folder's result has the default name depending on fitness function and datetime. So, I prefer users use the set_filename method if they want a shorter name. The example is `env.set_filename('GAQASPrepareW')`; then this line changed to `best_circuit = utilities.load_circuit("./GAQASPrepareW/best_circuit.qpy")`